### ISSUE TITLE
Possible NPE in Image Component

### ISSUE THREAD

**Issue by zajcu** [adobe/aem-core-wcm-components#928]
## Bug Report

**Current Behavior**
Context: a project is proxying the Image component and wants to set up or get areas from the model. 
When image asset is not yet filled (for example because there was an initial component Drag&Drop). 
Then ImageImpl throws NPE.

**Expected behavior/code**
Line https://github.com/adobe/aem-core-wcm-components/blob/master/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java#L182 is not null safe.

**Possible Solution**
Set up an empty list of areas as a callback or add nullchecker.

**Environment**
- AEM 6.5+
- Core Components version 2.8.0+
- JRE version 1.8.0+

**Additional Context**
The issue occurs as regression in the project after upgrade to use Core Component in version 2.8.0

**Comment by vladbailescu** [comment]
Thanks for reporting the issue. Proposed fix in https://github.com/adobe/aem-core-wcm-components/pull/930

**Comment by vladbailescu** [comment]
Actually, it seems `findbugs` is smarter than me:

```
[INFO] Redundant nullcheck of com.adobe.cq.wcm.core.components.internal.models.v2.ImageImpl.areas, which is known to be non-null in com.adobe.cq.wcm.core.components.internal.models.v2.ImageImpl.getAreas() [com.adobe.cq.wcm.core.components.internal.models.v2.ImageImpl] Redundant null check at ImageImpl.java:[line 182] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
```

`areas` is initialised at https://github.com/adobe/aem-core-wcm-components/blob/master/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java#L145 and cannot be null at the end of the method.

@zajcu, I'm interested to understand how to reproduce the issue you're facing, could you provide the simplest steps?

**Comment by zajcu** [comment]
Hi @vladbailescu , thank you to take a look on it.

To reproduce
Code part:
1. create simple MyImage component by proxying Image component where sling:resourceSuperType="core/wcm/components/image/v2/image"
2. create own MyImage sling model by implementing com.adobe.cq.wcm.core.components.models.Image interface
3. inject in MyImage original CC image model by use : 
@Self
  @Via(type = ResourceSuperType.class)
  private Image imageSuperTypeModel;

4. In MyImage @PostConstruct method invoke:
imageSuperTypeModel.getAreas()

Usage part:
1. Drag&Drop a new myImage component in to parsys

Actual result: 
Stack trace about NPE in com.adobe.cq.wcm.core.components.internal.models.v2.ImageImpl

Expected result:
No stack trace, empty state for component is valid

Best regards

**Comment by zajcu** [comment]
stack trace
`Caused by: org.apache.sling.scripting.sightly.SightlyException: Identifier com.some.package.ImageModel cannot be correctly instantiated by the Use API
	at org.apache.sling.scripting.sightly.impl.engine.extension.use.UseRuntimeExtension.call(UseRuntimeExtension.java:77) [org.apache.sling.scripting.sightly:1.1.2.1_4_0]
	at org.apache.sling.scripting.sightly.impl.engine.runtime.RenderContextImpl.call(RenderContextImpl.java:77) [org.apache.sling.scripting.sightly:1.1.2.1_4_0]
	at org.apache.sling.scripting.sightly.apps.some.package.image.image_html.render(image_html.java:43)
	at org.apache.sling.scripting.sightly.render.RenderUnit.render(RenderUnit.java:50) [org.apache.sling.scripting.sightly.runtime:1.1.0.1_4_0]
	at org.apache.sling.scripting.sightly.impl.engine.SightlyCompiledScript.eval(SightlyCompiledScript.java:60) [org.apache.sling.scripting.sightly:1.1.2.1_4_0]
	at org.apache.sling.scripting.core.impl.DefaultSlingScript.call(DefaultSlingScript.java:386) [org.apache.sling.scripting.core:2.0.56]
	at org.apache.sling.scripting.core.impl.DefaultSlingScript.eval(DefaultSlingScript.java:184) [org.apache.sling.scripting.core:2.0.56]
	at org.apache.sling.scripting.core.impl.DefaultSlingScript.service(DefaultSlingScript.java:491) [org.apache.sling.scripting.core:2.0.56]
	... 132 common frames omitted
Caused by: org.apache.sling.models.factory.PostConstructException: Post-construct method has thrown an exception for model class com.some.package.ImageModel
	at org.apache.sling.models.impl.ModelAdapterFactory.createObject(ModelAdapterFactory.java:774) [org.apache.sling.models.impl:1.4.10]
	at org.apache.sling.models.impl.ModelAdapterFactory.internalCreateModel(ModelAdapterFactory.java:448) [org.apache.sling.models.impl:1.4.10]
	at org.apache.sling.models.impl.ModelAdapterFactory.createModel(ModelAdapterFactory.java:314) [org.apache.sling.models.impl:1.4.10]
	at org.apache.sling.scripting.sightly.models.impl.SlingModelsUseProvider.provide(SlingModelsUseProvider.java:126) [org.apache.sling.scripting.sightly.models.provider:1.0.8]
	at org.apache.sling.scripting.sightly.impl.engine.extension.use.UseRuntimeExtension.call(UseRuntimeExtension.java:72) [org.apache.sling.scripting.sightly:1.1.2.1_4_0]
	... 139 common frames omitted
Caused by: java.lang.NullPointerException: null
	at java.util.Collections$UnmodifiableCollection.<init>(Unknown Source)
	at java.util.Collections$UnmodifiableList.<init>(Unknown Source)
	at java.util.Collections.unmodifiableList(Unknown Source)
	at com.adobe.cq.wcm.core.components.internal.models.v2.ImageImpl.getAreas(ImageImpl.java:145)
	at com.some.package.ImageModel.getAreas(ImageModel.java:171) [com.some.package:0.2.1.rc1-SNAPSHOT]
	at com.some.package.ImageModel.init(ImageModel.java:95) [com.some.package.core:0.2.1.rc1-SNAPSHOT]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.apache.sling.models.impl.ModelAdapterFactory.invokePostConstruct(ModelAdapterFactory.java:972) [org.apache.sling.models.impl:1.4.10]
	at org.apache.sling.models.impl.ModelAdapterFactory.createObject(ModelAdapterFactory.java:772) [org.apache.sling.models.impl:1.4.10]
	... 143 common frames omitted`

**Comment by vladbailescu** [comment]
It seems this was recently fixed in https://github.com/adobe/aem-core-wcm-components/commit/0e6b45d82712dcf48e11fcd575af2c7a7754ea83#diff-561deebea47b85bc8ffddf78fe11602d and this is a duplicate of #887

**Comment by zajcu** [comment]
You are right, regression with NPE is in 2.8.0 where the code is:
 @Override
    public List<ImageArea> getAreas() { return Collections.unmodifiableList(areas); }

but already fixed in 2.9.0, so I'm waiting for the next release, thanks.
